docs/wiki/The library - Module design.md

```diff
@@ -317,7 +317,7 @@ module <mainResource>_privateEndpoints '../../Microsoft.Network/privateEndpoints
     subnetResourceId: privateEndpoint.subnetResourceId
     enableDefaultTelemetry: enableDefaultTelemetry
     location: reference(split(privateEndpoint.subnetResourceId, '/subnets/')[0], '2020-06-01', 'Full').location
-    lock: contains(privateEndpoint, 'lock') ? privateEndpoint.lock : 'NotSpecified'
+    lock: contains(privateEndpoint, 'lock') ? privateEndpoint.lock : lock
```

Comment:
Its not really regarding this line, but should we add suggestion on how locks should be treated for child recourses in a section or example here?

As they inherit the locks, the default should be non-invasive and code should be single source of truth, so children only inherit locks, but locks can be enabled by supplying their own using the `lock` property/parameter of the child resource.

For tightly connected resources, the default is that these get locked based on the locks on the depending resource.

Reply:
Same comment like above:

> @MariusStorhaug I agree for child resources and will check the locks for that. However, what you pointed out here is not a child. Wouldn't we want to lock Endpoints and e.g. PublicIPs for VMs too?

Reply:
This is more that the decision and implementation is captured. We decided something, our guidance should reflect it.

Reply:
Correct. Hence I made sure that child-resources don't have locks passed down - and all others have

Reply:
Again, we should document this decision. That's all I ask for here.
- Child and extension resources: We do not automatically pass lock down, as locks are inherited by default in Azure.
- Cross-referenced resources: All cross-referenced resources are locked with the main resource to prevent depending resources to be changed or deleted.

Reply:
Updated the Module-Design `Locks` pattern description: https://github.com/Azure/ResourceModules/blob/users/alsehr/1445_locksArray/docs/wiki/The%20library%20-%20Module%20design.md#patterns

Reply:
👍👍👍